numpy/meson.build

```diff
@@ -10,15 +10,20 @@ endif
 
 # Platform detection
 is_windows = host_machine.system() == 'windows'
-is_mingw = is_windows and cc.get_id() == 'gcc'
+is_mingw = is_windows and cc.get_id() in ['gcc', 'clang']
+is_mingw_python = is_mingw and run_command(
+  py, ['-c', 'import sysconfig; print(sysconfig.get_platform())'],
+  check: true).stdout().strip().startswith('mingw')
```

Comment:
This variable is only used within the next block, so can you move it into `if `is_mingw`? That avoids running it when we already know that we're not using Mingw.

Reply:
Also, could you change the name to `is_mingw_built_python`, to make it a little clearer?

Reply:
It's already guarded by `is_mingw`, but sure, sounds good.

Reply:
I've renamed the variable and moved it into the suggested block.